src/component/directive.ts

```diff
@@ -428,7 +425,10 @@ QWeb.addDirective({
           const key = slotNode.getAttribute("t-set")!;
           slotNode.removeAttribute("t-set");
           const slotFn = qweb._compile(`slot_${key}_template`, slotNode, ctx);
-          QWeb.slots[`${slotId}_${key}`] = slotFn;
+          QWeb.slots[`${slotId}_${key}`] = function ctxSlotFn(context, extra) {
+            context = ctx.currentComponent ? ctx.currentComponent : context;
```

Comment:
I am afraid this is wrong (but previous code was wrong as well). This means that the compiled function depends on the rendering context, which is the first component rendered.  I am pretty sure that this means that if the same template is rendered with a different component, you will get a wrong component